pkg/benchmark/throughput.go

```diff
@@ -0,0 +1,187 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package benchmark
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onsi/gomega"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func StartThroughputTests(intraCluster, verbose bool) {
+	var f *framework.Framework
+
+	if verbose {
+		fmt.Printf("Performing throughput tests\n")
+	}
+
+	gomega.RegisterFailHandler(func(message string, callerSkip ...int) {
+		if f != nil {
+			cleanupFramework(f)
+		} else {
+			framework.RunCleanupActions()
+		}
+		panic(message)
+	})
+
+	f = initFramework("throughput")
+
+	clusterAName := framework.TestContext.ClusterIDs[framework.ClusterA]
+
+	if !intraCluster {
+		testParams := benchmarkTestParams{
+			ClientCluster:       framework.ClusterA,
+			ServerCluster:       framework.ClusterB,
+			ServerPodScheduling: framework.GatewayNode,
+			ClientPodScheduling: framework.GatewayNode,
+		}
+
+		clusterBName := framework.TestContext.ClusterIDs[framework.ClusterB]
+		fmt.Printf("Performing throughput tests from Gateway pod on cluster %q to Gateway pod on cluster %q\n",
+			clusterAName, clusterBName)
+		runThroughputTest(f, testParams, verbose)
+
+		testParams.ServerPodScheduling = framework.NonGatewayNode
+		testParams.ClientPodScheduling = framework.NonGatewayNode
+
+		fmt.Printf("Performing throughput tests from Non-Gateway pod on cluster %q to Non-Gateway pod on cluster %q\n",
+			clusterAName, clusterBName)
+		runThroughputTest(f, testParams, verbose)
+	} else {
+		testIntraClusterParams := benchmarkTestParams{
+			ClientCluster:       framework.ClusterA,
+			ServerCluster:       framework.ClusterA,
+			ServerPodScheduling: framework.GatewayNode,
+			ClientPodScheduling: framework.NonGatewayNode,
+		}
+
+		fmt.Printf("Performing throughput tests from Non-Gateway pod to Gateway pod on cluster %q\n", clusterAName)
+		runThroughputTest(f, testIntraClusterParams, verbose)
+	}
+
+	cleanupFramework(f)
+}
+
+var By = func(str string) {
+	fmt.Println(str)
+}
+
+func initFramework(baseName string) *framework.Framework {
+	f := framework.NewBareFramework(baseName)
+	framework.SetStatusFunction(By)
```

Comment:
Instead of making `By` global and checking the `verbose` flag everywhere, you can define the function here and check the verbose flag.

```suggestion
	framework.SetStatusFunction(func(str string) {
	     if verbose {
	         fmt.Println(str)
	     }
         })
```